sympy/combinatorics/rewritingsystem_fsm.py

```diff
@@ -0,0 +1,66 @@
+class State(object):
+    '''
+    A representation of a state managed by a ``StateMachine``.
+
+    Attributes:
+        name (instance of FreeGroupElement or string) -- State name which is also assigned to the Machine.
+        transisitons (OrderedDict) -- Represents all the transitions of the state object.
+        state_type (string) -- Denotes the type (accept/start/dead) of the state.
+        rh_rule (instance of FreeGroupElement) -- right hand rule for dead state.
+        state_machine (instance of StateMachine object) -- The finite state machine that the state belongs to.
+    '''
+
+    def __init__(self, name, state_machine, state_type=None, rh_rule=None):
+        self.name = name
+        self.transitions = {}
+        self.state_machine = state_machine
+        self.state_type = state_type
+        self.rh_rule = rh_rule
+        if self.state_type == "start":
+            self.state_type = 's'
+        elif self.state_type == "accept":
+            self.state_type = 'a'
+        elif self.state_type == "dead":
+            self.state_type = 'd'
```

Comment:
I think may have missed a part of my comment from last time: this can be more compactly written as `self.state_type = state_type[0]`, whatever the user's input.